### PR TITLE
Reduce overview sandbox mode notice footprint

### LINKED ISSUE #8449: WooPayments post-onboarding enhancements to improve activation
### Description
The current Payments Overview experience is overwhelming for the new user upon completing onboarding. There are three competing CTAs and announcements that aren't relevant to the user at this critical stage. We have a larger scope of work ongoing to provide better education for the new user post-onboarding (cc @pierorocca) 

These two small wins will improve the onboarding and activation experience for users immediately. It will facilitate merchant focus on the most important task of completing their store and payments setup.

### Acceptance criteria

* Change sandbox mode notice CTA to a text link.
* ~~Delay the appearance of the Fraud tools announcement for 3 days after connection.~~ There is another issue about this: https://github.com/Automattic/woocommerce-payments/issues/8279 

### Designs
Frame 1000005936

### Testing instructions

1. Create a new store and complete WooPayments onboarding and verification
2. See Payments Overview experience

cc @timmy5685

### PR DESCRIPTION
**Pull request by vladolaru**
Fixes #8449 

#### Changes proposed in this Pull Request

Update the sandbox mode notice on the Payments Overview page for a smaller footprint:
- use an inline link for opening the modal to switch to a live account
- use a tooltip to provide information about sandbox mode and link to the docs

I've also added Tracks events for the learn more links.

Before:

After:
Screenshot 2024-04-12 at 13 27 38

Screenshot 2024-04-12 at 13 27 47

#### Testing instructions

* Checkout the PR branch and build the client assets but running `npm run build:client`
* Make sure you don't have dev mode enabled via the WCPay Dev Tools
* Onboard a new account and choose the "I'm setting up the store for someone else" option on the Payments Connect page
* Go to the Payments Overview page and the sandbox mode notice should look like in the screenshots.
* Open up the Tracks Vigilante browser extension and make sure that there is a `wcadmin_wcpay_overview_sandbox_mode_learn_more_clicked` event fired when clicking on the "Learn more" link in the tooltip.
* Click on the switch to live link and click on "Continue setup" to switch to a live account
* You should be taken to our MOX wizard
* Finish onboarding and the notice should not be shown on the Payments Overview page

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8631` or branch name `update/8449-change-overview-sandbox-mode-notice` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=update/8449-change-overview-sandbox-mode-notice) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 82837f21bd999df26df4bff5ee375caa099d41f2
- Build time: 2024-04-12 12:35:54 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: changelog/update-8449-change-overview-sandbox-mode-notice
Comment (anu-rock) [inline changelog/update-8449-change-overview-sandbox-mode-notice:4]: I really like crispness of this short message.

File: client/overview/index.js
Comment (anu-rock) [inline client/overview/index.js]: The notice is spanning too many lines in the Overview page file. IMO, we should extract it as a separate component and import it back here, like `TestModeNotice`.

File: client/overview/index.js
Comment (anu-rock) [inline client/overview/index.js]: Interesting. I have never seen the `interpolateComponents` technique. It's like reading Flutter/Dart's declarative code 👍

**Review by anu-rock — COMMENTED** [review]
Good work here 👏 I've left one minor comment.

___

#### Review checklist

| Parameter  | Result |
| ------------- | ------------- |
| Approach  | ✅ All good |
| Implementation  | 👍 Almost good (left a comment about component extraction) |
| Automated tests  | ✅ All good |
| Naming  | ✅ All good |
| Typing  | ✅ All good |
| Code comments  | ✅ All good |
| Changelog  | ✅ All good |
| Testing instructions  | ✅ All good |
| Demo (video or JN site) or screenshots  | ✅ All good |
| Manual testing  | ⚠️  Skipped |

**Review by mordeth — APPROVED** [review]
✅ Code changes looks good!
✅ Notice and tooltip are displayed correctly on Desktop.
✅ Notice and tooltip are displayed correctly on Mobile.

Screenshot 2024-04-12 at 14 24 45

LGTM! :shipit:

File: client/overview/index.js
Comment (vladolaru) [reply client/overview/index.js]: I thought about it but, as with PHP abstractions, I think we should first have the need to use this in multiple places before abstracting it. As it stands, this notice style is only for the overview page. That is why I left it inline - I don't like having one-time-use components.

Also, if we abstract, this would be a special case of the `TestModeNotice` and the logic there would get quite complicated to support this level of flexibility. 

WDYT?

**Review by dpaun1985 — APPROVED** [review]
✅  LGTM
Code changes  looks good.
I tested and works as expected.

File: client/overview/index.js
Comment (anu-rock) [reply client/overview/index.js]: That's a fair argument. I am usually not pedantic about having all components in separate files. However, in this case, it is embedded inside a ternary operator, making it hard to see the alternate case without collapsing it in the code editor. Besides, it's the only compound component in the file that is this way, making things look a little inconsistent. WDYT about extracting it in the same file, like `OverviewPageError`?

That said, I'll leave it up to you as it's a matter of preference.

**Review by anu-rock — APPROVED** [review]
Pre-approving with an unresolved minor comment.

File: client/overview/index.js
Comment (vladolaru) [reply client/overview/index.js]: @anu-rock You do make a good argument. I will extract that code in the same file, in a function.

**Pull request merged**
